src/main/java/core/basesyntax/WorkWithFile.java

```
@@ -1,7 +1,49 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
     public void getStatistic(String fromFileName, String toFileName) {
+        String buyer = "buy";
+        String supplier = "supply";
+        int buy = 0;
+        int supply = 0;
 
+        File file = new File(fromFileName);
+        try {
+            BufferedReader readLine = new BufferedReader(new FileReader(file));
+            String value = readLine.readLine();
+            while (value != null) {
+                if (value.contains(supplier)) {
+                    int index = value.indexOf(",") + 1;
+                    String name = value.substring(index, value.length());
+                    supply += Integer.parseInt(name);
+                } else if (value.contains(buyer)) {
+                    int index = value.indexOf(",") + 1;
+                    String name = value.substring(index, value.length());
+                    buy += Integer.parseInt(name);
+                }
+                value = readLine.readLine();
+            }
+        } catch (FileNotFoundException e) {
+            throw new RuntimeException("Can`t read document",e);
```

The FileNotFoundException catch rethrows a RuntimeException with a very generic message: `throw new RuntimeException("Can`t read document", e);`. The checklist requires rethrowing exceptions with a meaningful message and the original exception as the cause ("Don't ignore exceptions"). Include the file name in the message, e.g., `"Can't read data from file " + fromFileName`.